### PR TITLE
 SOV-68: Implemented Tooltips

### PR DESCRIPTION
**Pull request by nsteward96**
#68 
* Added Tooltips
* Added TooltipBuilder. This takes a hash of optional parameters and returns a tooltip built around those parameters.

### CODE REVIEW

File: Client/Public/CSS/default.css [hunk 1]
```diff
@@ -164,7 +165,7 @@ body, html {
 }
 
 .job-description-box, .building-description-box {
-    height: 125px;
+    /*height: 125px;*/
```
Comment (nsteward96) [inline Client/Public/CSS/default.css]: NTS-1: Remove commented code

File: Client/Public/JS/Controllers/GameController.js [hunk 2]
```diff
@@ -86,19 +87,24 @@ function initModels() {
     // Init the jobs model.
     modelJobs['resourceCollector'] = 
         {
-            id: document.getElementById('jobResourceCollector').id,
-            allocateButton: document.getElementById('jobResourceCollector').children[2],
-            deallocateButton: document.getElementById('jobResourceCollector').children[3]
+            id: 'jobResourceCollector',
+            resourceType: ['resource'],
+            velocity: {'resource': modelResourceRates['resourceCollector'] * modelResource['townspeopleResourceCollector']},
+            allocateButton: 'resourceCollectorAddWorker',
```
Comment (nsteward96) [inline Client/Public/JS/Controllers/GameController.js]: NTS-2: Probably rename these 'allocateButtonId' and 'deallocateButtonId'. And any other place that specifies an id.

**Comment by thaibucher** [comment]
Great work exile, you hunt well! Einhar will take this task to the menagerie.

File: Client/Public/JS/Controllers/GameController.js [hunk 3]
```diff
@@ -204,6 +217,62 @@ function outputToFlavorTextArea(text) {
...
+        };
+    };
+    
+    var tooltips = document.getElementsByClassName('tooltip-container');
+    for (let i = 0; i < tooltips.length; i++) {
+        tooltips[i].parentElement.removeEventListener('mouseenter', displayTooltip(tooltips[i]));
+        tooltips[i].parentElement.addEventListener('mouseenter', displayTooltip(tooltips[i]));
+        tooltips[i].parentElement.removeEventListener('mouseout', hideTooltip(tooltips[i]));
+        tooltips[i].parentElement.addEventListener('mouseout', hideTooltip(tooltips[i]));
+        // $(tooltips[i].parentElement).off('mouseenter').on('mouseenter', displayTooltip(tooltips[i]));
```
Comment (nsteward96) [inline Client/Public/JS/Controllers/GameController.js]: NTS-3: Comments

File: Client/Public/CSS/default.css (hunk 1 above)
Comment (nsteward96) [reply Client/Public/CSS/default.css]: NTS-1: Removed

File: Client/Public/JS/Controllers/GameController.js (hunk 3 above)
Comment (nsteward96) [reply Client/Public/JS/Controllers/GameController.js]: NTS-3: Removed comments

File: Client/Public/JS/Controllers/GameController.js [hunk 4]
```diff
@@ -204,6 +217,60 @@ function outputToFlavorTextArea(text) {
     $(messageJustAppended).fadeIn(350);
 }
 
+function createTooltips() {
```
Comment (nsteward96) [inline Client/Public/JS/Controllers/GameController.js]: NTS-4: Reorganize functions so they are in the proper place in the JavaScript in terms of execution.

**Comment by nsteward96** [comment]
NTS-4: Reorganize functions so they are in the proper place in the JavaScript in terms of execution.

**Pull request merged**